components/scream/src/physics/shoc/shoc_clipping_diag_third_shoc_moments_impl.hpp

```diff
@@ -0,0 +1,35 @@
+#ifndef SHOC_CLIPPING_DIAG_THIRD_SHOC_MOMENTS_IMPL_HPP
+#define SHOC_CLIPPING_DIAG_THIRD_SHOC_MOMENTS_IMPL_HPP
+
+#include "shoc_functions.hpp" // for ETI only but harmless for GPU
+
+namespace scream {
+namespace shoc {
+
+template<typename S, typename D>
+KOKKOS_FUNCTION
+void Functions<S,D>
+::clipping_diag_third_shoc_moments(
+  const MemberType& team,
+  const Int& nlevi,
+  const uview_1d<const Spack>& w_sec_zi,
+  const uview_1d<Spack>& w3)
+{
+  const Int nlevi_pack = ekat::pack::npack<Spack>(nlevi);
+
+  Kokkos::parallel_for(Kokkos::TeamThreadRange(team, nlevi_pack), [&] (const Int& k) {
+    const auto w3clip = scream::shoc::Constants<Scalar>::w3clip;
+    Spack tsign(1.0);
```

Comment:
It's harmless here, and in the -1.0 below, but it's worth getting in the habit of using integer literals when you can.